src/descriptor/pegin/dynafed_pegin.rs

```diff
@@ -94,14 +94,17 @@ impl_from_tree!(
         if top.name == "pegin" && top.args.len() == 2 {
             // a roundtrip hack to use FromTree from bitcoin::Miniscript from
             // expression::Tree in elements.
-            let ms_str = top.args[0].to_string();
-            let ms_expr = BtcTree::from_str(&ms_str)?;
+            // let ms_str = top.args[0].to_string();
+            // let ms_expr = BtcTree::from_str(&ms_str)?;
             //
             // TODO: Confirm with Andrew about the descriptor type for dynafed
             // Assuming sh(wsh) for now.
-            let fed_desc = BtcDescriptor::<Pk>::from_tree(&ms_expr)?;
-            let elem_desc = Descriptor::<Pk, CovenantExt<CovExtArgs>>::from_tree(&top.args[1])?;
-            Ok(Pegin::new(fed_desc, elem_desc))
+            // let fed_desc = crate::BtcDescriptor::<Pk>::from_tree(&ms_expr)?;
+
+            // let fed_desc = BtcDescriptor::<Pk>::from_tree(&ms_expr)?;
```

Comment:
this line fails because `variant or associated item cannot be called on `Descriptor<Pk>` due to unsatisfied trait bounds` and I don't get what I can do about that

Reply:
In the `impl_from_tree` macro we have a ton of bounds on `Pk`. Upstream these have all been wrapped up into the [FromStrKey trait](https://docs.rs/miniscript/latest/miniscript/trait.FromStrKey.html) which you need to update the macro to use instead.

If you change this line to call `bitcoin_miniscript::expression::FromTree::from_tree` you will get a much better error message  from the compiler.

Reply:
Done in https://github.com/ElementsProject/elements-miniscript/pull/90/commits/e6724fcd29a1411df0bcf52c6a56fc3fe0417fde